includes/corrections/class-corrections.php

```diff
@@ -0,0 +1,503 @@
+			'item_updated'             => __( 'Correction updated.', 'newspack-plugin' ),
+			'item_link'                => __( 'Correction Link', 'newspack-plugin' ),
+			'item_link_description'    => __( 'A link to a correction.', 'newspack-plugin' ),
+		];
+		$args = array(
+			'labels'           => $labels,
+			'description'      => 'Post type used to store corrections and clarifications.',
+			'has_archive'      => true,
+			'public'           => true,
+			'public_queryable' => true,
+			'query_var'        => true,
+			'rewrite'          => [ 'slug' => 'correction' ],
+			'show_ui'          => true,
+			'show_in_rest'     => true,
+			'supports'         => $supports,
+			'taxonomies'       => [],
+			'menu_icon'        => 'dashicons-edit',
+		);
+		\register_post_type( self::POST_TYPE, $args );
+	}
+
+	/**
+	 * Get corrections for post.
+	 *
+	 * @param int $post_id The post ID.
+	 *
+	 * @return array The corrections.
+	 */
+	public static function get_corrections( $post_id ) {
+		$correction_ids = get_post_meta( $post_id, self::CORRECTIONS_IDS_META, true );
+		if ( ! is_array( $correction_ids ) ) {
+			return [];
+		}
+		return get_posts(
+			[
+				'posts_per_page' => -1,
+				'post_type'      => self::POST_TYPE,
+				'include'        => $correction_ids,
+			]
+		);
+	}
+
+	/**
+	 * Save corrections for post.
+	 *
+	 * @param int   $post_id     The post ID.
+	 * @param array $corrections The corrections.
+	 */
+	public static function save_corrections( $post_id, $corrections ) {
+		$correction_ids = get_post_meta( $post_id, self::CORRECTIONS_IDS_META, true );
+		if ( ! is_array( $correction_ids ) ) {
+			$correction_ids = [];
+		}
+		foreach ( $corrections as $correction ) {
+			$id = wp_insert_post(
+				[
+					'post_title'   => 'Correction for ' . get_the_title( $post_id ),
+					'post_content' => $correction['content'],
+					'post_type'    => self::POST_TYPE,
+					'post_status'  => 'publish',
+					'meta_input'   => [
+						self::CORRECTION_POST_ID_META => $post_id,
+						self::CORRECTION_DATE_META    => $correction['date'],
+					],
+				]
+			);
+			if ( ! \is_wp_error( $id ) ) {
+				$correction_ids[] = $id;
+			}
+		}
+		update_post_meta( $post_id, self::CORRECTIONS_IDS_META, $correction_ids );
+	}
+
+	/**
+	 * Update correction.
+	 *
+	 * @param int   $correction_id the post id.
+	 * @param array $correction    the correction.
+	 */
+	public static function update_correction( $correction_id, $correction ) {
+		wp_update_post(
+			[
+				'ID'           => $correction_id,
+				'post_content' => sanitize_textarea_field( $correction['content'] ),
+			]
+		);
+		update_post_meta( $correction_id, self::CORRECTION_DATE_META, sanitize_text_field( $correction['date'] ) );
+	}
+
+	/**
+	 * Delete corrections for post.
+	 *
+	 * @param array $correction_ids correction ids.
+	 */
+	public static function delete_corrections( $correction_ids ) {
+		$stored_correction_ids = get_post_meta( $post_id, self::CORRECTIONS_IDS_META, true );
```

Comment:
`$post_id` doesn't seem to be defined here

Reply:
Oops! Good catch. Fixed in https://github.com/Automattic/newspack-plugin/pull/3638/commits/fea98909ba23a62584c9f97bd3a2adc97bf1be2e